sherpa/astro/ui/tests/test_astro_ui.py

```diff
@@ -428,7 +429,11 @@ def save_arrays(self):
         rtol = 0
         atol = 1e-5
         self.assertIsInstance(out, Data1D)
-        self.assertEqual(out.name, ofh.name, msg="file name")
+
+        # remove potential dm syntax introduced by backend before checking for equality
+        out_name = re.sub("\[.*\]", "", out.name)
```

Comment:
I think this should be anchored to the end of the string - e.g. `"\[.*\]$"` - but is it worth it? I guess first I should check that `re` supports this!